src/goto-analyzer/goto_analyzer_parse_options.cpp

```diff
@@ -56,6 +56,25 @@ goto_analyzer_parse_optionst::goto_analyzer_parse_optionst(
 {
 }
 
+void goto_analyzer_parse_optionst::set_default_analysis_flags(optionst &options)
+{
+  // Checks enabled by default in v6.0+.
+  options.set_option("bounds-check", true);
+  options.set_option("pointer-check", true);
+  options.set_option("pointer-primitive-check", true);
+  options.set_option("div-by-zero-check", true);
+  options.set_option("signed-overflow-check", true);
+  options.set_option("undefined-shift-check", true);
```

Comment:
The `PARSE_OPTIONS_GOTO_CHECK_NEGATIVE_DEFAULT_CHECKS` is there to *overwrite* the flags in case they happen to be different (say, if a user is using default checks, but also `--no-div-by-zero-check`).

I agree that the end result is setting the flags to `true` twice, but I like that they are present in `set_default_analysis_flags` as documentation of the tool's default behaviour at the very least, and also without having to *depend* on flags being *implicitly* set to the default values by inverted logic in a different place in the code (`goto_check_c.h`).